homepage/homepage/content/docs/using-covalues/version-control.mdx

```diff
@@ -0,0 +1,246 @@
+      tasks: { $each: true }
+    },
+    unstable_branch: {
+      name: currentBranchName,
+    },
+  });
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    // Won't be visible on main until merged
+    project.$jazz.set("title", e.target.value);
+  };
+
+  const handleTaskTitleChange = (index: number, e: React.ChangeEvent<HTMLInputElement>) => {
+    const task = project.tasks[index];
+
+    // The task is also part of the branch because we used the `resolve` option
+    // with `tasks: { $each: true }`
+    // so the changes won't be visible on main until merged
+    task?.$jazz.set("title", e.target.value);
+  };
+
+  return (
+    <form onSubmit={handleSave}>
+      {/* Edit form fields */}
+    </form>
+  );
+}
+```
+</CodeGroup>
+</ContentByFramework>
+<ContentByFramework framework={["svelte"]}>
+<CodeGroup>
+```svelte
+<script lang="ts">
+  import { CoState } from 'jazz-tools/svelte';
+  import { Project } from './schema.js';
+
+  let projectId = $state<string>();
+  let currentBranchName = $state<string>('main');
+
+  const project = new CoState(Project, () => projectId, () => ({
+    resolve: {
+      tasks: { $each: true }
+    },
+    unstable_branch: currentBranchName === 'main' ? undefined : { name: currentBranchName }
+  }));
+
+  function handleTitleChange(e: Event) {
+    const target = e.target as HTMLInputElement;
+    // Won't be visible on main until merged
+    project.current?.$jazz.set("title", target.value);
+  }
+
+  function handleTaskTitleChange(index: number, e: Event) {
+    const target = e.target as HTMLInputElement;
+    const task = project.current?.tasks[index];
+
+    // The task is also part of the branch because we used the `resolve` option
+    // with `tasks: { $each: true }`
+    // so the changes won't be visible on main until merged
+    task?.$jazz.set("title", target.value);
+  }
+</script>
+
+<form>
+  <!-- Edit form fields -->
+  <input 
+    type="text" 
+    value={project.current?.title || ''} 
+    oninput={handleTitleChange}
+  />
+  
+  {#each project.current?.tasks || [] as task, index}
+    <input 
+      type="text" 
+      value={task.title || ''} 
+      oninput={(e) => handleTaskTitleChange(index, e)}
+    />
+  {/each}
+</form>
+```
+</CodeGroup>
+</ContentByFramework>
+
+### Merging
+
+To merge a branch back to the main version, call `unstable_merge()`:
+
+<CodeGroup>
+```tsx 
+function handleSave() {
+  // Merge the branch changes back to main
+  project.$jazz.unstable_merge();
+  router.navigate("/");
+}
+```
+</CodeGroup>
+
+All the changes made to the branch will be merged into the main CoValue, preserving both author and timestamp.
+
+Calling `unstable_merge()` will merge every change done in the project and the loaded tasks.
+
+The merge is idempotent, so you can merge the same branch multiple times without changing the result.
+
+The merge operation is scoped, so if you call `unstable_merge()` on a task, only the changes to the task will be merged
```

Comment:
Calling `unstable_merge()` on the project won't update branched task fields either, right? I think this may be worth a more prominent callout